Migrating abp packages to standalone structure

erdemcaygor opened the pull request:
### Description

Resolves https://github.com/volosoft/volo/issues/19845

### Account Module: Standalone Migration

#### 🧱 Module Structure
- [x] Account module converted to standalone setup
- [x] Routes exported as `accountRoutes` using Angular standalone routing
- [x] Routing uses `provideRouter()` or is consumed via `loadChildren: () => import(...).then(m => m.accountRoutes)`

#### 🧩 Component Standalone Migration
- [x] ChangePasswordComponent migrated to standalone
- [x] ForgotPasswordComponent migrated to standalone
- [x] LoginComponent migrated to standalone
- [x] ManageProfileComponent migrated to standalone
- [x] PersonalSettingsComponent migrated to standalone
- [x] RegisterComponent migrated to standalone
- [x] ResetPasswordComponent migrated to standalone

###  Identity Module: Standalone Migration

#### 🧱 Module Structure
- [x] Identity module converted to standalone setup
- [x] Routes exported as `identityRoutes` using Angular standalone routing
- [x] Routing uses `provideRouter()` or is consumed via `loadChildren: () => import(...).then(m => m.identityRoutes)`

#### 🧩 Component Standalone Migration
- [x] RolesComponent migrated to standalone
- [x] UsersComponent migrated to standalone

###  Tenant Management Module: Standalone Migration

#### 🧱 Module Structure
- [x] Tenant management module converted to standalone setup
- [x] Routes exported as `tenantManagementRoutes` using Angular standalone routing
- [x] Routing uses `provideRouter()` or is consumed via `loadChildren: () => import(...).then(m => m.tenantManagementRoutes)`

#### 🧩 Component Standalone Migration
- [x] TenantsComponent migrated to standalone

###  Setting Management Module: Standalone Migration

#### 🧱 Module Structure
- [x] Setting management module converted to standalone setup
- [x] Routes exported as `settingManagementRoutes` using Angular standalone routing
- [x] Routing uses `provideRouter()` or is consumed via `loadChildren: () => import(...).then(m => m.settingManagementRoutes)`

#### 🧩 Component Standalone Migration
- [x] SettingComponent migrated to standalone

###  Core Module: Standalone Migration

#### 🧱 Module Structure
- [x] Core module converted to standalone setup

#### 🧩 Component Standalone Migration
- [x] Core components, directives, pipes migrated to standalone

###  Theme Basic Module: Standalone Migration

#### 🧱 Module Structure
- [x] Theme basic module converted to standalone setup

#### 🧩 Component Standalone Migration
- [x] Theme basic components, directives, pipes migrated to standalone

###  Theme Shared Module: Standalone Migration

#### 🧱 Module Structure
- [x] Theme shared module converted to standalone setup

#### 🧩 Component Standalone Migration
- [x] Theme shared components, directives, pipes migrated to standalone

### How to Test

Use `yarn run copy-to:app` command

#### Change Theme

To replace the theme, run:

```bash
npx ng g @abp/ng.schematics:change-theme

erdemcaygor committed:
standalone structure applied to login component

erdemcaygor committed:
standalone structure applied to register component

erdemcaygor committed:
standalone structure applied to forgot password component

erdemcaygor committed:
standalone applied to reset password, personal settings and change password

erdemcaygor committed:
refactoring

erdemcaygor committed:
account standalone module and router created

erdemcaygor committed:
refactoring

erdemcaygor committed:
identity module standalone migration

erdemcaygor committed:
identity module standalone migration

erdemcaygor committed:
tenant management module standalone migration

erdemcaygor committed:
setting management module standalone migration

erdemcaygor committed:
refactoring

erdemcaygor committed:
refactoring

erdemcaygor committed:
packages ng-modules refactored

erdemcaygor committed:
theme basic module standalone migration

erdemcaygor committed:
theme basic module standalone migration

erdemcaygor committed:
removed unused modules

erdemcaygor committed:
refactoring

erdemcaygor committed:
teheme shared module standalone migration

erdemcaygor committed:
theme shared module standalone migration

erdemcaygor committed:
theme shared module standalone migration

erdemcaygor committed:
core module standalone migration

erdemcaygor committed:
core module standalone migration

erdemcaygor committed:
refactoring

erdemcaygor committed:
feature management module standalone migration

erdemcaygor committed:
refactoring

erdemcaygor committed:
fix

erdemcaygor committed:
type fix

erdemcaygor committed:
Merge branch 'dev' into feature/migrating-abp-packages-to-standalone-structure

# Conflicts:
#	npm/ng-packs/packages/core/src/lib/core.module.ts
#	npm/ng-packs/packages/core/src/lib/localization.module.ts
#	npm/ng-packs/packages/theme-basic/src/lib/theme-basic.module.ts

erdemcaygor committed:
merge from dev

erdemcaygor committed:
refactoring

erdemcaygor committed:
refactoring

erdemcaygor committed:
fix

erdemcaygor committed:
test updates

erdemcaygor committed:
test updates

erdemcaygor committed:
refactoring

erdemcaygor committed:
tenant management module refactored

erdemcaygor committed:
setting management module refactored

erdemcaygor committed:
fix

erdemcaygor committed:
Merge branch 'dev' into feature/migrating-abp-packages-to-standalone-structure

erdemcaygor committed:
dev app standalone migration

erdemcaygor committed:
dev app standalone migration

erdemcaygor committed:
refactoring

erdemcaygor committed:
dev app config updated

erdemcaygor committed:
refactoring

erdemcaygor committed:
refactoring

masum-ulu commented on npm/ng-packs/apps/dev-app/src/app/app.component.ts:
```diff
@@ -1,12 +1,21 @@
     <abp-loader-bar></abp-loader-bar>
     <abp-dynamic-layout></abp-dynamic-layout>
     <abp-internet-status></abp-internet-status>
+    <router-outlet />
```
Remove if not need it

erdemcaygor replied:
removed

masum-ulu commented on npm/ng-packs/apps/dev-app/src/app/app.component.ts:
```diff
@@ -1,12 +1,21 @@
+    LoaderBarComponent,
+    DynamicLayoutComponent,
+    InternetConnectionStatusComponent,
+    RouterOutlet,
```
Remove if not need it

erdemcaygor replied:
removed

masum-ulu reviewed (changes requested):
## Revisions

- [x] Change `<package-name>.config.ts` name to `<package-name>.routes.ts` (update public-api.ts file in packages)
- [x] Update *.routes.ts content like below (e.g of identity package)
```ts
import { Provider } from '@angular/core';
import { Routes } from '@angular/router';

import {
  authGuard,
  permissionGuard,
  ReplaceableComponents,
  ReplaceableRouteContainerComponent,
  RouterOutletComponent,
} from '@abp/ng.core';
import {
  IDENTITY_CREATE_FORM_PROP_CONTRIBUTORS,
  IDENTITY_EDIT_FORM_PROP_CONTRIBUTORS,
  IDENTITY_ENTITY_ACTION_CONTRIBUTORS,
  IDENTITY_ENTITY_PROP_CONTRIBUTORS,
  IDENTITY_TOOLBAR_ACTION_CONTRIBUTORS,
} from './tokens';
import { IdentityConfigOptions } from './models';
import { RolesComponent, UsersComponent } from './components';
import { identityExtensionsResolver } from './resolvers';
import { eIdentityComponents } from './enums';

function provideIdentityConfig(options: IdentityConfigOptions = {}): Provider[] {
  return [
    {
      provide: IDENTITY_ENTITY_ACTION_CONTRIBUTORS,
      useValue: options.entityActionContributors,
    },
    {
      provide: IDENTITY_TOOLBAR_ACTION_CONTRIBUTORS,
      useValue: options.toolbarActionContributors,
    },
    {
      provide: IDENTITY_ENTITY_PROP_CONTRIBUTORS,
      useValue: options.entityPropContributors,
    },
    {
      provide: IDENTITY_CREATE_FORM_PROP_CONTRIBUTORS,
      useValue: options.createFormPropContributors,
    },
    {
      provide: IDENTITY_EDIT_FORM_PROP_CONTRIBUTORS,
      useValue: options.editFormPropContributors,
    },
  ];
}

export const createRoutes = (options: IdentityConfigOptions = {}): Routes => [
  {
    path: '',
    component: RouterOutletComponent,
    canActivate: [authGuard, permissionGuard],
    resolve: [identityExtensionsResolver],
    providers: provideIdentityConfig(options),
    children: [
      { path: '', redirectTo: 'roles', pathMatch: 'full' },
      {
        path: 'roles',
        component: ReplaceableRouteContainerComponent,
        data: {
          requiredPolicy: 'AbpIdentity.Roles',
          replaceableComponent: {
            key: eIdentityComponents.Roles,
            defaultComponent: RolesComponent,
          } as ReplaceableComponents.RouteData<RolesComponent>,
        },
        title: 'AbpIdentity::Roles',
      },
      {
        path: 'users',
        component: ReplaceableRouteContainerComponent,
        data: {
          requiredPolicy: 'AbpIdentity.Users',
          replaceableComponent: {
            key: eIdentityComponents.Users,
            defaultComponent: UsersComponent,
          } as ReplaceableComponents.RouteData<UsersComponent>,
        },
        title: 'AbpIdentity::Users',
      },
    ],
  },
];
```
- [ ] Update app.routes.ts like below in dev-app
```ts
 {
    path: 'identity',
    loadChildren: () => import('@abp/ng.identity').then(m => m.createRoutes()),
},
```

masum-ulu commented:
Some coponent directives and pipes still have `standalone: true` definition no need it with ng 19
![image](https://github.com/user-attachments/assets/27199deb-6fba-4ced-9f8b-5517d756c1d7)

masum-ulu reviewed (changes requested):
Some components, directives and pipes still have `standalone: true` definition no need it with ng 19
![image](https://github.com/user-attachments/assets/27199deb-6fba-4ced-9f8b-5517d756c1d7)

erdemcaygor committed:
refactoring

erdemcaygor committed:
refactoring

erdemcaygor committed:
refactoring

erdemcaygor commented:
> ## Revisions
> * [x]  Change `<package-name>.config.ts` name to `<package-name>.routes.ts` (update public-api.ts file in packages)
> * [x]  Update *.routes.ts content like below (e.g of identity package)
> 
> ```ts
> import { Provider } from '@angular/core';
> import { Routes } from '@angular/router';
> 
> import {
>   authGuard,
>   permissionGuard,
>   ReplaceableComponents,
>   ReplaceableRouteContainerComponent,
>   RouterOutletComponent,
> } from '@abp/ng.core';
> import {
>   IDENTITY_CREATE_FORM_PROP_CONTRIBUTORS,
>   IDENTITY_EDIT_FORM_PROP_CONTRIBUTORS,
>   IDENTITY_ENTITY_ACTION_CONTRIBUTORS,
>   IDENTITY_ENTITY_PROP_CONTRIBUTORS,
>   IDENTITY_TOOLBAR_ACTION_CONTRIBUTORS,
> } from './tokens';
> import { IdentityConfigOptions } from './models';
> import { RolesComponent, UsersComponent } from './components';
> import { identityExtensionsResolver } from './resolvers';
> import { eIdentityComponents } from './enums';
> 
> function provideIdentityConfig(options: IdentityConfigOptions = {}): Provider[] {
>   return [
>     {
>       provide: IDENTITY_ENTITY_ACTION_CONTRIBUTORS,
>       useValue: options.entityActionContributors,
>     },
>     {
>       provide: IDENTITY_TOOLBAR_ACTION_CONTRIBUTORS,
>       useValue: options.toolbarActionContributors,
>     },
>     {
>       provide: IDENTITY_ENTITY_PROP_CONTRIBUTORS,
>       useValue: options.entityPropContributors,
>     },
>     {
>       provide: IDENTITY_CREATE_FORM_PROP_CONTRIBUTORS,
>       useValue: options.createFormPropContributors,
>     },
>     {
>       provide: IDENTITY_EDIT_FORM_PROP_CONTRIBUTORS,
>       useValue: options.editFormPropContributors,
>     },
>   ];
> }
> 
> export const createRoutes = (options: IdentityConfigOptions = {}): Routes => [
>   {
>     path: '',
>     component: RouterOutletComponent,
>     canActivate: [authGuard, permissionGuard],
>     resolve: [identityExtensionsResolver],
>     providers: provideIdentityConfig(options),
>     children: [
>       { path: '', redirectTo: 'roles', pathMatch: 'full' },
>       {
>         path: 'roles',
>         component: ReplaceableRouteContainerComponent,
>         data: {
>           requiredPolicy: 'AbpIdentity.Roles',
>           replaceableComponent: {
>             key: eIdentityComponents.Roles,
>             defaultComponent: RolesComponent,
>           } as ReplaceableComponents.RouteData<RolesComponent>,
>         },
>         title: 'AbpIdentity::Roles',
>       },
>       {
>         path: 'users',
>         component: ReplaceableRouteContainerComponent,
>         data: {
>           requiredPolicy: 'AbpIdentity.Users',
>           replaceableComponent: {
>             key: eIdentityComponents.Users,
>             defaultComponent: UsersComponent,
>           } as ReplaceableComponents.RouteData<UsersComponent>,
>         },
>         title: 'AbpIdentity::Users',
>       },
>     ],
>   },
> ];
> ```
> 
> * [x]  Update app.routes.ts like below in dev-app
> 
> ```ts
>  {
>     path: 'identity',
>     loadChildren: () => import('@abp/ng.identity').then(m => m.createRoutes()),
> },
> ```

Completed: Renamed *.config.ts to *.routes.ts, updated public-api.ts, and adjusted route file contents.

erdemcaygor committed:
unnecessary <router-outlet> removed

erdemcaygor committed:
app template migrated to standalone structure

erdemcaygor committed:
app nolayer template migrated to standalone structure

erdemcaygor committed:
module template migrated to standalone structure

sumeyyeKurtulus commented on npm/ng-packs/packages/theme-shared/src/lib/components/password/password.component.ts:
```diff
@@ -1,15 +1,15 @@
 @Component({
   selector: 'abp-password',
-  imports: [CommonModule, FormsModule],
+  imports: [CommonModule, FormsModule, ReactiveFormsModule],
```
`NgxValidateCoreModule` is missing in these imports.

sumeyyeKurtulus commented on npm/ng-packs/packages/account/src/lib/components/change-password/change-password.component.ts:
```diff
@@ -1,26 +1,27 @@
   selector: 'abp-change-password-form',
   templateUrl: './change-password.component.html',
   exportAs: 'abpChangePasswordForm',
+  imports: [ReactiveFormsModule, LocalizationPipe, ButtonComponent, NgxValidateCoreModule],
```
`AutoFocusDirective` is missing in these imports.

sumeyyeKurtulus commented on npm/ng-packs/packages/account/src/lib/components/login/login.component.ts:
```diff
@@ -1,18 +1,35 @@
-  standalone: false,
   selector: 'abp-login',
   templateUrl: './login.component.html',
+  imports: [
```
`AutoFocusDirective` is missing in these imports.

sumeyyeKurtulus commented on npm/ng-packs/packages/account/src/lib/components/register/register.component.ts:
```diff
@@ -1,19 +1,32 @@
-  standalone: false,
   selector: 'abp-register',
   templateUrl: './register.component.html',
+  imports: [
```
`AutoFocusDirective` is missing in these imports.

sumeyyeKurtulus reviewed (changes requested):
These components still use our modules such as `CoreModule` , `ThemeSharedModule`, `PageModule` etc. It is better to replace them with relative standalone imports.
- account
    - [x]  `personal-settings-half-row.component.ts`
- identity
    - [x]  `roles.component.ts`
    - [x]  `users.component.ts`
- setting-management
    - [x]  `setting-management.component.ts`
- theme-basic
    - [x]  `auth-wrapper.component.ts`
    - [x]  `tenant-box.component.ts`
    - [x]  `account-layout.component.ts`
    - [x]  `application-layout.component.ts`
    - [x]  `current-user.component.ts`
    - [x]  `languages.component.ts`
    - [x]  `nav-items.component.ts`
    - [x]  `page-alert-container.component.ts`
    - [x]  `routes.component.ts`
    - [x]  `validation-error.component.ts`
    
- [x] We can also use the `createRoutes()` function in the templates.
- [x] Adding deprecation marks for the functions that we already provided in the route creation would be better.

erdemcaygor committed:
refactoring

sumeyyeKurtulus reviewed (approved):
Ready for testing 🚀 @oykuermann 🙌🏻

oykuermann reviewed (changes requested):
# App Template LeptonX Lite

- [ ] Tenant management > Features Cancel button does not work:
<img width="1020" alt="image" src="https://github.com/user-attachments/assets/a085ec4c-b36b-4f55-8025-c60c0ed16df9" />

- [ ] Settings > Feature Management Cancel button does not work:
![image](https://github.com/user-attachments/assets/44907c63-bcf2-410f-aff4-427973885634)

# App Template Basic 

- [ ] How can I log out from my user? There is no button. How can I go Manage account page? 
![image](https://github.com/user-attachments/assets/b1f2d8db-1733-40c5-8d1e-0294d901b53a)

- [ ] Cancel button on Feature management modal is not working on Basic theme too.

oykuermann commented:
# App-nolayers LeptonX Lite

- [ ] My Account page is not opening:

[screen-capture - 2025-05-22T151420.838.webm](https://github.com/user-attachments/assets/21aaa725-d3dd-4164-81ab-02784aff0bf0)

+ app template mistakes

erdemcaygor committed:
Merge branch 'dev' into feature/migrating-abp-packages-to-standalone-structure

erdemcaygor committed:
fix

erdemcaygor committed:
fix

erdemcaygor committed:
fix
